v3/src/components/slider/slider-component-handler.ts

```diff
@@ -0,0 +1,202 @@
+import { SetRequired } from "type-fest"
+import { V2Slider } from "../../data-interactive/data-interactive-component-types"
+import { DIComponentHandler } from "../../data-interactive/handlers/component-handler"
+import { errorResult } from "../../data-interactive/handlers/di-results"
+import { appState } from "../../models/app-state"
+import { GlobalValueManager } from "../../models/global/global-value-manager"
+import { t } from "../../utilities/translation/translate"
+import { kSliderTileType, kV2SliderType } from "./slider-defs"
+import { ISliderModel, ISliderSnapshot, isSliderModel } from "./slider-model"
+import { AnimationDirections, AnimationModes, SliderScaleTypes } from "./slider-types"
+import { applySnapshot, getSnapshot } from "mobx-state-tree"
+import { cloneDeep } from "lodash"
+import { dateUnits } from "../../utilities/date-utils"
+import { DIValues } from "../../data-interactive/data-interactive-types"
+
+function validateInput(values: DIValues | undefined, sliderModel?: ISliderModel): {
+  error?: ReturnType<typeof errorResult>
+  validatedValues?: Partial<ISliderSnapshot>
+} {
+  if (!values) {
+    return {validatedValues: {}}
+  }
+  const {
+    animationDirection: _animationDirection,
+    animationMode: _animationMode,
+    globalValueName,
+    scaleType,
+    dateMultipleOfUnit,
+  } = values as V2Slider
+
+  const snapshot: Partial<ISliderSnapshot> = {}
+
+  if (globalValueName) {
+    const { document } = appState
+    const globalManager = document.content?.getFirstSharedModelByType(GlobalValueManager)
+    const global = globalManager?.getValueByName(globalValueName)
+    if (!global) {
+      return {error: errorResult(t("V3.DI.Error.globalNotFound", { vars: [globalValueName] }))}
+    }
+    snapshot.globalValue = global.id
+
+    // If this is a new slider, or the global is being changed, check for multiple sliders
+    if (!sliderModel || sliderModel.globalValue.id !== global.id) {
+      // Multiple sliders for one global value are not allowed
+      let existingTile = false
+      document.content?.tileMap.forEach(sliderTile => {
+        if (isSliderModel(sliderTile.content) && sliderTile.content.globalValue.id === global.id) {
+          existingTile = true
+        }
+      })
+      if (existingTile) {
+        return {error: errorResult(t("V3.DI.Error.noMultipleSliders", { vars: [globalValueName] }))}
+      }
+    }
+  }
+
+  if (dateMultipleOfUnit != null) {
+    if (!dateUnits.includes(dateMultipleOfUnit as any)) {
+      return {error: errorResult(t("V3.DI.Error.unsupportedDateUnit", { vars: [dateMultipleOfUnit] }))}
+    }
+    snapshot.dateMultipleOfUnit = dateMultipleOfUnit as ISliderSnapshot["dateMultipleOfUnit"]
```

Comment:
I feel like there must be a way to do this where it's not necessary to cast variables like this and in the conditional below. But maybe it's more work to set that up than it's worth.

Reply:
I think the way to clean this up, including the `any` a few lines above, is to implement an `isDateUnit()` type guard function which takes a string and return `value is DateUnit`.

Reply:
Yeah I can do that. It adds more lines of code and still requires a cast to do the `includes` check inside of `isDateUnit()`, but it does look nicer. And perhaps the isDateUnit will come in handy some other time.

Reply:
True, but you can cast the other way: `(dateUnits as string[]).includes(value)` which seems nicer because it's a widening cast instead of a narrowing one, i.e. it assumes less.

Reply:
I found a better way to do it, but I already merged the PR.

The `(dateUnits as string[]).includes(value)` isn't better because the user could pass almost anything as value since it is JSON object from a plugin in this case.

So the best approach seems to be:
```
export function isDateUnit(value: any): value is DateUnit {
  return dateUnits.includes(value)
}
```

When I ended up merging was:
```
export function isDateUnit(value: any): value is DateUnit {
  return dateUnits.includes(value as DateUnit) 
```
That `as DateUnit` is unnecessary.